PersonListener/Gateway/TenureInfoApiGateway.cs

```diff
@@ -47,11 +47,12 @@ private static JsonSerializerOptions CreateJsonOptions()
         }
 
         [LogCall]
-        public async Task<TenureResponseObject> GetTenureInfoByIdAsync(Guid id)
+        public async Task<TenureResponseObject> GetTenureInfoByIdAsync(Guid id, Guid correlationId)
         {
             var client = _httpClientFactory.CreateClient();
             var getTenureRoute = $"{_getTenureApiRoute}/tenures/{id}";
 
+            client.DefaultRequestHeaders.Add("x-correlation-id", correlationId.ToString());
```

Comment:
My thought is whether it's worth creating a client via the nuget package project. I am guessing this will be the approach with all our APIs

Reply:
Yeah... The ApiGateway class here could very easily be made generic and put into a package somewhere. I was going to do this when I got the time...